### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by BorzdeG**
- Update to detekt 1.18.1
- Update to Gradle 7.2
- Update to IntelliJ 2021.2.1
- Update Gradle plugins
- Using deprecated code will generate an error during verification plugin
- Update to JVM 11

### CODE REVIEW

**Comment by cortinico** [comment]
Thank for your PR @BorzdeG. We will hold this on till we have a stable release of `1.18.0`

**Comment by BorzdeG** [comment]
@cortinico Yes, I made a draft PR - as soon as a stable release comes out, I will make a full PR

**Review by cortinico — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -12,33 +15,41 @@ repositories {
 }
 
 plugins {
-    id("org.jetbrains.intellij").version("0.7.3")
-    id("com.github.ben-manes.versions") version "0.38.0"
-    kotlin("jvm").version("1.4.32")
+    id("org.jetbrains.intellij") version "1.1.4"
+    id("com.github.ben-manes.versions") version "0.39.0"
+    kotlin("jvm") version embeddedKotlinVersion
```
Comment (cortinico) [inline build.gradle.kts:20]: Let's keep the `id("...").version("...")` notation for all of those.

File: build.gradle.kts [hunk 2]
```diff
@@ -12,33 +15,41 @@ repositories {
...
     runtimeOnly("io.gitlab.arturbosch.detekt:detekt-rules:$detektVersion")
     runtimeOnly("io.gitlab.arturbosch.detekt:detekt-formatting:$detektVersion")
+
     testImplementation("io.gitlab.arturbosch.detekt:detekt-test-utils:$detektVersion")
-    testImplementation("org.assertj:assertj-core:3.19.0")
+    testImplementation("org.assertj:assertj-core:3.20.2")
     testImplementation("org.junit.jupiter:junit-jupiter:5.7.2")
 }
 
+val jvmVersion = JavaVersion.VERSION_11
```
Comment (cortinico) [inline build.gradle.kts:37]: Why are we bumping this?

**Review by cortinico — COMMENTED** [review]
Thanks for taking care of this 🙏

File: build.gradle.kts (hunk 2 above)
Comment (BorzdeG) [reply build.gradle.kts:37]: 1. in GitHub Flow, the check went exactly in this version
2. minimum version of IDEA (2020.3.4) uses exactly this version of JVM
3. gradle-intellij-plugin only works with this version

I think that you need to build the plugin with the minimum version of IDEA and check compatibility for older versions

**Review by cortinico — APPROVED** [review]

**Pull request merged**